src/Serializer/ZddMessageMessengerSerializer.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Yousign\ZddMessageBundle\Serializer;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface as MessengerSerializerInterface;
+
+class ZddMessageMessengerSerializer implements SerializerInterface
+{
+    private MessengerSerializerInterface $serializer;
+
+    public function __construct(MessengerSerializerInterface $serializer)
+    {
+        $this->serializer = $serializer;
+    }
+
+    public function serialize(mixed $data): string
+    {
+        if (!\is_object($data)) {
+            throw new \InvalidArgumentException(sprintf('Object expected, %s provided', \gettype($data)));
+        }
+
+        return \json_encode($this->serializer->encode(Envelope::wrap($data)), JSON_THROW_ON_ERROR);
```

Comment:
You don't need to encore the result, juste take the `body` of the array return by the method `encode` of the messenger serializer